language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -293,7 +293,7 @@ public ContextTypeResolver(PositionedOperationContext context) {
         if (typeRefSymbol.isEmpty()) {
             return Optional.empty();
         }
-        
+
```

Comment:
I think still the sorting order is not finalized. We get the following, where `b` should be at the top. Are we having the context type resolving issue here for the langlib methods?
<img width="647" alt="Screen Shot 2022-03-24 at 8 56 17 AM" src="https://user-images.githubusercontent.com/1329674/159836052-21b8a21b-7bec-4953-848a-f655660153da.png">

Reply:
For some reason https://github.com/ballerina-platform/ballerina-lang/blob/02411da0cf01ac4f76c768ed911728b0e0ad115b/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MethodCallExpressionNodeContext.java#L111 we have decided to rank the methods/functions first. Should we change this behavior. Is there a specific reason behind this @IMS94

Reply:
@malinthar Seems like the logic should be improved. If we are within the parameter context, we should rank `symbolKind.VARIABLE` higher. Adding a new `else if` condition for `Variables` will do the trick IMO.
```java
                    } else if (symbol.stream().anyMatch(sym -> sym.kind() == SymbolKind.VARIABLE)) {
                        rank = withinParameterCtx ? 1 : 3;
                        break;
                    } else {
```

Reply:
Or we can SortingUtil.toRank() method which ranks the completion items in the standard order

Reply:
My concern here is, since the main method returns `()` while b is already an int variable, why we are not showing the variable `b` at the top.

Reply:
@nadeeshaan We haven't considered assignability when sorting here. That's the issue. If we consider assignability, then this issue will be sorted automatically.

Reply:
I have fixed the sorting order as follows. 
![image](https://user-images.githubusercontent.com/35211477/160328176-ae2ac2f0-3215-44a4-ac30-64c8caa6452a.png)

Reply:
Can we move the `arr` to bottom? What's the context type we get here? `int[]` or `int`?

Reply:
https://github.com/ballerina-platform/ballerina-lang/blob/95f08123f94dafc0e1a7758917f6f6822b6c83c1/langlib/lang.array/src/main/ballerina/array.bal#L245

As depicted above the first parameter of the push method is of type Type[]. Hence, sorting `arr` to the top is correct in this case IMO.

Reply:
@malinthar not really. What we are looking at here is method access where the method is coming from a standard library. For the standard libraries, the first parameter is being ignored if the types of the first parameter and the expression's type are the same. Hence we have to revamp this sorting